### ISSUE TITLE
[Request] Ada Chaperone as an ES6 Module

### ISSUE THREAD

**Issue by martinmckenna** [AdaSupport/docs#5]
I'm currently using Ada in a production React app. I would love to eventually be able to import Ada as an ES6 module (hopefully with TypeScript support)

Currently, here's how I'm implementing Ada now

```js
export class MyComponent extends React.Component<Props, State> {
  state: State = {};

  ada: any = undefined;

  componentDidMount() {
    /*
    * Init Ada Chaperone chat app
    * Script is included in index.html
    */
    this.ada = new (window as any).AdaChaperone('my_key');
  }

  handleAdaInit = () => {
    /*
    * Show the Ada chat
    */
    this.ada.show();
  }

  render() {
    const { classes } = this.props;

    return (
      <button onClick={this.handleAdaInit} />
    )
  }
}
```

**Comment by NicHaley** [comment]
Hey @martinmckenna. We have no immediate plans to support ES6 modules in Chaperone (but we would certainly like to). Will update if/when this changes

**Comment by martinmckenna** [comment]
thanks. When this does get addressed, I would also love type definitions along with it :)

**Comment by batmot** [comment]
Hi ! 

2 years passed and still no support for ES6 modules ?

**Comment by NicHaley** [comment]
@batmot Unfortunately no -- there have not been many requests for it, so it has been hard to prioritize! That being said, we _should_ offer this. I appreciate the renewed interest and will see if I can get the ball rolling!

**Comment by batmot** [comment]
Yeah I understand, it should often be used with no code platform.

With new create-react-app, we don't have access anymore to the index.html file.
Webpack externals may do the trick 🤔
And I need to wrap the initialize with launchDarkly.

Will be way way more obvious with ES6 module
